paasta_tools/kubernetes_tools.py

```diff
@@ -4325,21 +4325,42 @@ def ensure_service_account(
     kube_client: KubeClient,
     k8s_role: Optional[str] = None,
 ) -> None:
+    role_annotation = "eks.amazonaws.com/role-arn"
     sa_name = get_service_account_name(iam_role, k8s_role)
 
-    if not any(
-        sa.metadata and sa.metadata.name == sa_name
-        for sa in get_all_service_accounts(kube_client, namespace)
-    ):
```

Comment:
i don't quite know why we did things this way rather than getting a single SA with `read_namespaced_service_account()`, but this seems like something we can fix in another PR so as not to scope-creep this one